File: 0075-sort-colors/0075-sort-colors.cpp
class Solution {
public:
    void sortColors(vector<int>& nums) {
        int n=nums.size();
        int j=0;
        for(int i=0;i<n;i++)
        {
            if(nums[i]==0)
            {
                swap(nums[j++],nums[i]);
            } 
        }
        for(int i=0;i<n;i++)
        {
            if(nums[i]==1)
            {
                swap(nums[j++],nums[i]);
            } 
        }
       
    }
};

File: 0075-sort-colors/0075-sort-colors.js
/**
 * @param {number[]} nums
 * @return {void} Do not return anything, modify nums in-place instead.
 */

var sortColors = function(nums)
{
    let n=nums.length;
    let l=0,mid=0,h=n-1;
    while(mid<=h)
        {
            if(nums[mid]==0)
                {
                    [nums[l],nums[mid]]=[nums[mid],nums[l]];
                    l++;
                    mid++;
                }
            else if(nums[mid]==1)
                {
                    mid++;
                }
            else
                {
                    [nums[mid],nums[h]]=[nums[h],nums[mid]];
                    h--;
                }
        }
};